io/slow5/slow5.go

```diff
@@ -0,0 +1,409 @@
+				newRead.Error = fmt.Errorf("Failed to convert digitisation '%s' to float on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.Digitisation = digitisation
+		case "offset":
+			offset, err := strconv.ParseFloat(values[valueIndex], 64)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert offset '%s' to float on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.Offset = offset
+		case "range":
+			nanoporeRange, err := strconv.ParseFloat(values[valueIndex], 64)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert range '%s' to float on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.Range = nanoporeRange
+		case "sampling_rate":
+			samplingRate, err := strconv.ParseFloat(values[valueIndex], 64)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert sampling_rate '%s' to float on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.SamplingRate = samplingRate
+		case "len_raw_signal":
+			lenRawSignal, err := strconv.ParseUint(values[valueIndex], 10, 64)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert len_raw_signal '%s' to float on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.LenRawSignal = lenRawSignal
+		case "raw_signal":
+			var rawSignals []int16
+			for rawSignalIndex, rawSignalString := range strings.Split(values[valueIndex], ",") {
+				rawSignal, err := strconv.ParseInt(rawSignalString, 10, 16)
+				if err != nil {
+					newRead.Error = fmt.Errorf("Failed to convert raw signal '%s' to int on line %d, signal index %d. Got error: %s", rawSignalString, parser.line, rawSignalIndex, err)
+				}
+				rawSignals = append(rawSignals, int16(rawSignal))
+			}
+			newRead.RawSignal = rawSignals
+		case "start_time":
+			startTime, err := strconv.ParseUint(values[valueIndex], 10, 64)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert start_time '%s' to uint on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.StartTime = startTime
+		case "read_number":
+			readNumber, err := strconv.ParseInt(values[valueIndex], 10, 32)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert read_number '%s' to int on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.ReadNumber = int32(readNumber)
+		case "start_mux":
+			startMux, err := strconv.ParseUint(values[valueIndex], 10, 8)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert start_mux '%s' to uint on line %d. Got error: %s", values[valueIndex], parser.line, err)
+			}
+			newRead.StartMux = uint8(startMux)
+		case "median_before":
+			medianBefore, err := strconv.ParseFloat(values[valueIndex], 64)
+			if err != nil {
+				newRead.Error = fmt.Errorf("Failed to convert median_before '%s' to float on line %d. Got error: %s", values[valueIndex], parser.line, err)
```

Comment:
Might be worth replacing the final `%s` formatter verb with `%w` for wrapping functionality- very small change and you give users extra error handling functionality.